### PR TITLE
ft(user-loggedin): implements the controller function to get all user…

### PR DESCRIPTION
**Pull request by TMDav007**
#### What does this PR do?
- This PR implements the GET all users request api 

#### Description of Task to be completed?
- This PR will create a user dummmy data that has users as an admin and as a user
- It will create a requestController class with the method getAllRequests to get all users request
- It will update the route index with the GET method for getAllRequests
- It will write the test and tested with mocha and coverage
- It will add travis file

#### Any background context you want to provide?
- None

#### What are the relevant Github Issues?
- None

#### Questions:
- None

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,11 @@
   "description": "This is a application that keep tracker of requests from users",
```
Comment (nebanat) [inline package.json]: Provide more information in this description..

File: server/dummyControllers/requestController.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import Request from './../dummyModels/request';
+import User from './../dummyModels/user';
+
+/**
+ * it is a class that control all request api;
+ */
+class requestController {
```
Comment (nebanat) [inline server/dummyControllers/requestController.js]: Class names should be in PascalCase not camelCase... refer to the Airbnb style guide

File: server/dummyControllers/requestController.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        error: true,
+        message: 'you dont have the priviledge for this request'
+      });
+    }
+    // get all request
+    if (Request) {
+      return res.status(200).json({
+        error: false,
+        message: 'success',
+        Request
```
Comment (nebanat) [inline server/dummyControllers/requestController.js]: is there a reason why this variable Request is in PascalCase? Again refer to Airbnb style guide. what's the rationale behind the `error: false`?

File: server/dummyControllers/requestController.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      });
+    }
+    // return an error if request not found
+    return res.status(404).json({
```
Comment (nebanat) [inline server/dummyControllers/requestController.js]: When writing code been explicit will save you time in the long run...use an else statement to be more explicit

File: server/dummyModels/request.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+const requests = [
+  {
+    id: 1,
+    name: 'Okafor-John',
+    request: 'Request to fix the AC',
+    requestDetails: 'I hdjdjjdjfjfjdhhddfhfsbs hbsjhskshdb hfhjsdbjfbhsd hsdjfjbfsdjdshfbdh',
```
Comment (nebanat) [inline server/dummyModels/request.js]: Why the jibberish? Provide real-world examples, not jibberish...imagine if you were presenting this to other non-tech stakeholders

File: package.json (hunk 1 above)
Comment (TMDav007) [reply package.json]: Ok. I will do that.

File: server/dummyControllers/requestController.js (hunk 2 above)
Comment (TMDav007) [reply server/dummyControllers/requestController.js]: Ok

File: server/dummyControllers/requestController.js (hunk 3 above)
Comment (TMDav007) [reply server/dummyControllers/requestController.js]: The reason why I used "error: false" is to let the user know that there is no error in the request, Request is the name I use to reference the database dummy data.

File: server/dummyControllers/requestController.js (hunk 4 above)
Comment (TMDav007) [reply server/dummyControllers/requestController.js]: ok. 
I will be more explicit

File: server/dummyModels/request.js (hunk 5 above)
Comment (TMDav007) [reply server/dummyModels/request.js]: Ok. I will do that. Thank you

File: server/dummyControllers/requestController.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  /**
+     * it GET all requests
+     * @param {string} req
+     * @param {string} res
+     * @returns {object} object
+     */
+  static getAllRequests(req, res) {
+    const admin = users.filter(user => user.role === 'admin');
+    // check if user is admin to get all request
+    if (!admin) {
```
Comment (nebanat) [inline server/dummyControllers/requestController.js]: Is there a reason why an admin is denied rights to view a user requests?

File: server/route/index.js [hunk 7]
```diff
@@ -1,3 +1,6 @@
+import RequestController from './../dummyControllers/requestController';
+
+const { getAllRequests } = RequestController;
 
 const routes = (app) => {
   app.get('', (req, res) =>
```
Comment (nebanat) [inline server/route/index.js:6]: where does this route to?

File: server/dummyControllers/requestController.js (hunk 6 above)
Comment (TMDav007) [reply server/dummyControllers/requestController.js]: The admin has the only right to get all requests of the users, the admin wasnt denied the right

File: server/route/index.js (hunk 7 above)
Comment (TMDav007) [reply server/route/index.js:6]: it welcome the users and gives directions on how to use the app. e.g on heroku, when you input the url like: https:// maintenance-tacker.herokuapp.com, you get a response telling you to add /api/v1/ to use the app.

File: server/dummyControllers/requestController.js (hunk 6 above)
Comment (nebanat) [reply server/dummyControllers/requestController.js]: That's not foresight enough, let's say in the future the app grow and admin can filter requests based on a user from the admin dashboard.  You have to either edit this function **(violates the open and close principle of SOLID)** or create another function which will just be repeating code **(violates Single responsibility principle of SOLID)**.

Comment (nebanat) [reply server/dummyControllers/requestController.js]: An addendum is the admin has the ultimate rights to do anything on the system.

Comment (TMDav007) [reply server/dummyControllers/requestController.js]: I dont understand what you mean by when the "app grew and admin can filter requests based on a user from the admin dashboard", pls can you shed more light.
What that line of code does is check through the user db, and check if the user making the request to get all request is an admin.

**Pull request merged**